superset/db_engine_specs/base.py

```diff
@@ -911,13 +912,18 @@ def column_datatype_to_string(
     ) -> str:
         """
         Convert sqlalchemy column type to string representation.
-        Can be overridden to remove unnecessary details, especially
-        collation info (see mysql, mssql).
+        By default removes collation and character encoding info to avoid unnecessarily
+        long datatypes.
 
         :param sqla_column_type: SqlAlchemy column type
         :param dialect: Sqlalchemy dialect
         :return: Compiled column type
         """
+        sqla_column_type = sqla_column_type.copy()
+        if hasattr(sqla_column_type, "collation"):
```

Comment:
This seems way cleaner that the regex approach. My view is if you're using regex you're probably doing something wrong.

Reply:
I agree, this seems obvious in retrospect 👍